torchchronos/download.py

```diff
@@ -1,40 +1,76 @@
+import os
 import shutil
 import tempfile
 import urllib.request
 import zipfile
 from pathlib import Path
-from typing import Optional
 
-from sktime.datasets._data_io import _download_and_extract, _list_available_datasets
+from sktime.datasets._data_io import _list_available_datasets
 from .typing import AnyPath
 
 
-def download_uea_ucr(extract_path: Optional[Path], dataset_name: str) -> None:
-    """This downloads the uea ucr dataset from sktime to the path extract_path."""
+from pathlib import Path
+import zipfile
+
+
+def _list_available_datasets(data_dir: AnyPath) -> list[str]:
+    datasets = []
+    for name in os.listdir(data_dir):
+        sub_dir = os.path.join(data_dir, name)
+        if os.path.isdir(sub_dir):
+            all_files = os.listdir(sub_dir)
+            if name + "_TRAIN.ts" in all_files and name + "_TEST.ts" in all_files:
+                datasets.append(name)
+    return datasets
+
+
+def download_uea_ucr(
+    extract_path: str | Path | None = None, dataset_name: str = "GunPoint"
```

Comment:
I would not have a default here. In partical applications, it is not very useful.
```suggestion
    extract_path: str | Path | None = None, dataset_name: str
```

Reply:
(You'd have to switch the argument order too.)

Reply:
I know, but I wanted to keep the order, but yes I can do this